osquery/tables/system/windows/intel_me.cpp

```diff
@@ -124,31 +125,65 @@ void getHECIDriverVersion(QueryData& results) {
                         nullptr);
 
   if (ret == 0) {
-    VLOG(1) << "Device IOCTL call failed with " << GetLastError();
+    auto last_error = GetLastError();
+    if (last_error == ERROR_GEN_FAILURE) {
+      VLOG(1) << "The driver is already in use by another client and can't be "
+                 "queried at this time";
+    } else {
+      VLOG(1) << "Device IOCTL call failed with " << last_error;
+    }
+
     CloseHandle(driver);
     return;
   }
 
+  if (response.version != 0x1) {
+    VLOG(1) << "Intel MEI version is unsupported";
+    return;
+  }
+
+  if (response.maxlen < kMinResponseSize) {
+    VLOG(1) << "Invalid maxlen size";
+    return;
+  } else if (response.maxlen != 0x1000) {
```

Comment:
What does 0x1000 represent?

Reply:
I wanted to add some kind of warning (without failing the query) in case the response parameters are different than what we tested (so that the user can report this in case the query output is wrong)